README.md

```diff
@@ -16,3 +32,155 @@ pip install cookiecutter
 ```sh
 cookiecutter gh:elixir-cloud-aai/cookiecutter-python
 ```
+
+> **Note**: If you're using just `cookiecutter`, manually remove
+  `update-template` job from `update.yaml` github action workflow file as
+  `cookiecutter` doesn't support autosync like cruft.
+
+## Configurations
+
+### Package manager - Poetry
+
+This project uses [Poetry][poetry-org] as a package manager. Check out the
+commands at the [official documentation][poetry-docs].
+
+### Linters and formatters - Ruff
+
+To lint and format `Python` code files, it uses [Ruff][ruff], the default
+configuration is set in the `pyproject.toml` file.
+
+```toml
+select = [
+  "B", # flake8-bugbear
+  "E", # pycodestyle
+  "F", # Pyflakes
+  "I", # isort
+  "PL", # pylint
+  "SIM", # flake8-simplify
+  "UP", # pyupgrade
+]
+```
+
+To configure it to your needs, refer to the [rules documentation][rudff-linter]
+, and for formatter configuration, refer to the
+[configuration documentation][ruff-formatter].
+
+### Spell checker - Typos
+
+If you want to ignore certain words, add them to the `pyproject.toml` file,
+under the `tool.typos.default.extend-words` key.
+
+```toml
+[tool.typos.default.extend-words]
+mke = 'mke'
+```
+
+For further configuration, refer to the [typos docs][typos-docs].
+
+### Static type checker - Mypy
+
+Change the configuration in `pyproject.toml` file, for further information refer to
+the [documentation][mypy-config].
+
+### Documentation - Sphinx
+
+Configuration for Sphinx is in the `docs/source/conf.py` file, for further info refer
+to the [Sphinx documentation][sphinx-docs]. The default configuration uses
+[furo theme][furo] and [ReadTheDocs][rtd] to host the documentation, you can
+change the configuration using `/docs/source/conf.py` and `.readthedocs.yml`
+file.
```

Comment:
I suppose this requires setting up at Read the Docs itself? Perhaps that should be documented here.